modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectLayoutResourceImpl.java

```diff
@@ -141,6 +143,20 @@ public ObjectLayout putObjectLayout(
 			Long objectLayoutId, ObjectLayout objectLayout)
 		throws Exception {
 
+		if (Validator.isNotNull(
+				objectLayout.getObjectDefinitionExternalReferenceCode())) {
+
+			ObjectDefinition objectDefinition =
+				_objectDefinitionLocalService.
+					fetchObjectDefinitionByExternalReferenceCode(
+						contextCompany.getCompanyId(),
+						objectLayout.
+							getObjectDefinitionExternalReferenceCode());
+
+			objectLayout.setObjectDefinitionId(
+				objectDefinition.getObjectDefinitionId());
+		}
+
```

Comment:
Hey @seltong!

Sorry about not giving the right instructions. So, For the put* methods that receives a parameter *Id, we should consider the id first and then check the externalReferenceCode, ok?

Reply:
I got confused. You're doing everything good here.